metrics/ci_integrations/test/cli/configured_parties/factory/configured_parties_factory_test.dart

```diff
@@ -0,0 +1,239 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:ci_integration/cli/config/model/raw_integration_config.dart';
+import 'package:ci_integration/cli/configured_parties/factory/configured_parties_factory.dart';
+import 'package:ci_integration/cli/parties/parties.dart';
+import 'package:ci_integration/cli/parties/supported_integration_parties.dart';
+import 'package:ci_integration/integration/interface/base/config/parser/config_parser.dart';
+import 'package:ci_integration/integration/interface/base/party/integration_party.dart';
+import 'package:ci_integration/integration/interface/destination/config/model/destination_config.dart';
+import 'package:ci_integration/integration/interface/destination/party/destination_party.dart';
+import 'package:ci_integration/integration/interface/source/config/model/source_config.dart';
+import 'package:ci_integration/integration/interface/source/party/source_party.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../test_util/destination_config_stub.dart';
+import '../../test_util/mock/config_parser_mock.dart';
+import '../../test_util/mock/parties_mock.dart';
+import '../../test_util/source_config_stub.dart';
+
+// ignore_for_file: avoid_redundant_argument_values
+
+void main() {
+  group("ConfiguredPartiesFactory", () {
+    const sourceConfigMap = {'source': 'config'};
+    const destinationConfigMap = {'destination': 'config'};
+
+    final rawConfig = RawIntegrationConfig(
+      sourceConfigMap: sourceConfigMap,
+      destinationConfigMap: destinationConfigMap,
+    );
+
+    final sourceConfig = SourceConfigStub();
+    final destinationConfig = DestinationConfigStub();
+
+    final sourceParties = PartiesMock<SourceParty>();
+    final destinationParties = PartiesMock<DestinationParty>();
+
+    final supportedParties = SupportedIntegrationParties(
+      sourceParties: sourceParties,
+      destinationParties: destinationParties,
+    );
+
+    final configuredPartiesFactory = ConfiguredPartiesFactory(
+      supportedParties: supportedParties,
+    );
+
+    final sourceParty = _SourcePartyMock();
+    final destinationParty = _DestinationPartyMock();
+
+    final sourceConfigParser = ConfigParserMock<SourceConfig>();
+    final destinationConfigParser = ConfigParserMock<DestinationConfig>();
+
+    void setupParties(
+      Parties parties,
+      Map<String, dynamic> configMap,
+      IntegrationParty party,
+      ConfigParser configParser,
+    ) {
+      when(parties.getParty(configMap)).thenReturn(party);
+      when(party.configParser).thenReturn(configParser);
+    }
+
+    PostExpectation<SourceConfig> whenParseSourceConfig() {
+      setupParties(
+        sourceParties,
+        sourceConfigMap,
+        sourceParty,
+        sourceConfigParser,
+      );
+      return when(sourceConfigParser.parse(sourceConfigMap));
```

Comment:
Maybe we can extract this line into the `setupParties` method too? If so, would it be better to rename the `setupParties` to be something like `whenParseConfig`?